Debug array alloc ccpp caps

climbfuji opened the pull request:
## Description

This PR and associated PRs for ccpp-framework and ccpp-physics
- enable debugging features in auto-generated CCPP physics caps
- cleanup variables for Ferrier-Aligo and RRTMGP in `GFS_typedefs.{F90,meta}`
- contain bugfixes for multigases array dimensions in `CCPP_typedefs.{F90,meta}`
- and contain additional bug fixes in CCPP metadata in `GFS_typedefs.{F90,meta}`

Also update the submodule pointer for GFDL_atmos_cubed_sphere to point to the head of dev/emc (contains a number of bug fixes for initializing tracers correctly, and an update to get a better error message from CCPP in case call to `ccpp_physics_run` for group 'fast_physics' fails).

### Issue(s) addressed

n/a

## Testing

See https://github.com/ufs-community/ufs-weather-model/pull/850

## Dependencies

https://github.com/NCAR/ccpp-physics/pull/749
https://github.com/NCAR/ccpp-framework/pull/404
https://github.com/NOAA-EMC/fv3atm/pull/407/files
https://github.com/noaa-psd/stochastic_physics/pull/48
https://github.com/ufs-community/ufs-weather-model/pull/850

climbfuji committed:
Update .gitmodules and submodule pointer for ccpp-physics for code review and testing

climbfuji committed:
Make real vegetation, slope and soil type integers, remove unnecessary interstitial arrays

climbfuji committed:
Update submodule pointer for ccpp-physics

climbfuji committed:
Update submodule pointer for ccpp-physics

climbfuji committed:
Add interstitial variables to save and restore vegetation/soil/slope types before/after surface physics

climbfuji committed:
Update submodule pointer for ccpp-physics

climbfuji committed:
Make vegetation/soil/slope save variables persistent so that they can be used in the _init phases

climbfuji committed:
Only allow nearest-neighbor interpolation method for integer data

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into vegetation_soil_slope_type_integer

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into vegetation_soil_slope_type_integer

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into vegetation_soil_slope_type_integer

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into vegetation_soil_slope_type_integer

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into vegetation_soil_slope_type_integer

climbfuji committed:
Update submodule pointer for ccpp-physics

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into vegetation_soil_slope_type_integer

climbfuji committed:
Remove legacy code from CMakeLists.txt

climbfuji committed:
Update submodule pointer for ccpp-physics

climbfuji committed:
Revert change to .gitmodules and update submodule pointer for ccpp-physics

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into HEAD

climbfuji committed:
Update .gitmodules for code review and testing

climbfuji committed:
Update .gitmodules and submodule pointers for GFDL_atmos_cubed_sphere, ccpp-framework and ccpp-physics for code review and testing

climbfuji committed:
Enable debugging features in auto-generated CCPP physics caps, clean up of variables for Ferrier-Aligo and RRTMGP in GFS_typedefs, bugfixes for multigases array dimensions, bug fixes in CCPP metadata in GFS_typedefs

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into debug-array-alloc-ccpp-caps

climbfuji committed:
Revert change for GFDL_atmos_cubed_sphere to .gitmodules and update submodule pointer for GFDL_atmos_cubed_sphere

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into debug-array-alloc-ccpp-caps

climbfuji committed:
Update submodule pointer for ccpp-physics

climbfuji committed:
Revert submodule pointer update for GFDL_atmos_cubed_sphere

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into debug-array-alloc-ccpp-caps

climbfuji committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/fv3atm into debug-array-alloc-ccpp-caps

junwang-noaa commented on ccpp/suites/suite_FV3_GFS_v16_thompson.xml.dom:
```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<suite name="FV3_GFS_v16_thompson" version="1">
```
Dom, do you want to add this file with name suite_FV3_GFS_v16_thompson.xml.dom ?

climbfuji replied:
Argh, no. Apologies, an oversight. I'll remove it. That can be done safely, because a suite ending with something else than `.xml` can't be used by any regression test anyway. Thanks for catching this.

climbfuji committed:
Revert .gitmodules and update submodule pointers for ccpp-framework and ccpp-physics; remove ccpp/suites/suite_FV3_GFS_v16_thompson.xml.dom

climbfuji commented:
Reverted `.gitmodules` and updated submodule pointers for ccpp-framework and ccpp-physics - hashes are correct. Removed `ccpp/suites/suite_FV3_GFS_v16_thompson.xml.dom`. Ready to merge.
